src/EditorFeatures/CSharpTest/Formatting/FormattingEngineTestBase.cs

```diff
@@ -172,10 +175,18 @@ internal static void AssertFormat(Workspace workspace, string expected, OptionSe
             Assert.Equal(expected, actual);
         }
 
-        protected static void AssertFormatWithView(string expectedWithMarker, string codeWithMarker, bool debugMode = false)
+        protected static void AssertFormatWithView(string expectedWithMarker, string codeWithMarker, Dictionary<PerLanguageOption<bool>, bool> changedOptions = null, bool debugMode = false)
```

Comment:
just to reduce typing on test, we could do

AssertFormatWithView(..., params (PerLang... option, bool value)[] options) and inside of the method

call AssertFormatWithView(... options.ToDictionary(v => v.option, v => v.value)

and each test, one can just do AssertFormatWithView(... (..IsCodeCleanupEnabled, true), (SortUsing, true))

Reply:
and since everyone will want code clean up on code clean up test, you can add AssertCodeCleanupWithView or something that internally always add IsCodeCleanup, true

Reply:
updated